web/src/test/java/org/springframework/security/web/authentication/logout/CookieClearingLogoutHandlerTests.java

```diff
@@ -61,6 +62,18 @@ public void configuredCookiesAreCleared() {
 		}
 	}
 
+	@Test
+	public void configuredCookieIsSecure() {
+		MockHttpServletResponse response = new MockHttpServletResponse();
+		MockHttpServletRequest request = new MockHttpServletRequest();
+		request.setSecure(true);
+		request.setContextPath("/app");
+		CookieClearingLogoutHandler handler = new CookieClearingLogoutHandler("my_cookie");
+		handler.logout(request, response, mock(Authentication.class));
+		assertThat(response.getCookies()).hasSize(1);
+		assertThat(response.getCookies()[0].getSecure()).isTrue();
+	}
+
```

Comment:
Please add a test checking that when the request is not secure, then the cookie is not secure.